### PR TITLE
 [TKW] Introduce AbsOp as an op for Quantized LLM and GenAI workload. The tkw.abs op is lowered into math.absf and math.absi MLIR dialect for float and integer respectively.

### LINKED ISSUE #322: Add support for tkw.abs
implement tkw.abs and it's lowering into math.absfOp or math.absIOp in math dialect of MLIR. Here is a sample of a similar task on reciprocal (https://github.com/iree-org/iree-turbine/commit/82852d13198be9f18f6a070b2762bdaf6e78c5f3).

### PR DESCRIPTION
**Pull request by NoumanAmir657**
[TKW] Introduce AbsOp as an op for Quantized LLM and GenAI workload. 
The tkw.abs op is lowered into math.absf and math.absi MLIR dialect for float and integer respectively

Fixes #322

### CODE REVIEW

File: lit_tests/kernel/wave/codegen.py [hunk 1]
```diff
@@ -1941,6 +1949,10 @@ def test(a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f16]):
         # Testing reciprocal
         # %[[ONES:.+]] = arith.constant dense<1.000000e+00> : vector<4xf16>
         # %[[RECIPROCAL:.+]] = arith.divf %[[ONES]], %[[EXP2]] : vector<4xf16>
+        
+        # Testing abs
+        # %[[ABS:.+]] = math.absf %[[RECIPROCAL]]
+        # %[[ABS:.+]] = math.absi
```
Comment (raikonenfnu) [inline lit_tests/kernel/wave/codegen.py:1955]: Can we renamed the floating abs to `%[[ABSF:.+]]` and int one to `%[[ABSI:.+]]`

File: lit_tests/kernel/wave/codegen.py [hunk 2]
```diff
@@ -1921,16 +1921,24 @@ def test_unary_lowerings():
     constraints += [tkw.WaveConstraint(N, BLOCK_N / 2)]
 
     @tkw.wave(constraints)
-    def test(a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f16]):
+    def test(
+        a: tkl.Memory[M, N, ADDRESS_SPACE, tkl.f16],
+        b: tkl.Memory[M, N, ADDRESS_SPACE, tkl.i32],
```
Comment (raikonenfnu) [inline lit_tests/kernel/wave/codegen.py:1926]: nice! thanks for adding the int case here :)

**Review by raikonenfnu — COMMENTED** [review]
Thanks for the quick turnaround and nice PR! I just have some minor comments.

File: lit_tests/kernel/wave/codegen.py (hunk 1 above)
Comment (NoumanAmir657) [reply lit_tests/kernel/wave/codegen.py]: Fixed.

Comment (raikonenfnu) [reply lit_tests/kernel/wave/codegen.py]: thanks!

**Comment by raikonenfnu** [comment]
@NoumanAmir657 can we update the commit/PR message to be more descriptive of what you did plus a bit of the motivation behind adding this op (i.e Useful Operations for Quantized LLM and GenAI workload) :)

**Comment by NoumanAmir657** [comment]
> @NoumanAmir657 can we update the commit/PR message to be more descriptive of what you did plus a bit of the motivation behind adding this op (i.e Useful Operations for Quantized LLM and GenAI workload) :)

I have added it. Let me know if any further changes needed. Thanks!

**Comment by raikonenfnu** [comment]
a couple more things:

1. It's failing in pre-commit, before pushing up every PR, try to do:
```
pre-commit run --all-files
```
This will make re-style all the files in the repo to ensure it pass the repo style guide!
2.
> I have added it. Let me know if any further changes needed. Thanks!

Can you try update it in PR/web github as well?

After the two above, we should be good to go! :)

**Comment by NoumanAmir657** [comment]
> a couple more things:
> 
>     1. It's failing in pre-commit, before pushing up every PR, try to do:
> 
> 
> ```
> pre-commit run --all-files
> ```
> 
> This will make re-style all the files in the repo to ensure it pass the repo style guide! 2.
> 
> > I have added it. Let me know if any further changes needed. Thanks!
> 
> Can you try update it in PR/web github as well?
> 
> After the two above, we should be good to go! :)

Done!

**Review by raikonenfnu — APPROVED** [review]

**Pull request merged**